src/interpreter/crypto.rs

```diff
@@ -1,37 +1,38 @@
 use super::Interpreter;
-use crate::consts::{MEM_MAX_ACCESS_SIZE, VM_MAX_RAM};
-use crate::crypto;
+use crate::consts::{MEM_MAX_ACCESS_SIZE, MIN_VM_MAX_RAM_USIZE_MAX, VM_MAX_RAM};
 use crate::error::RuntimeError;
 
+use crate::arith::{checked_add_word, checked_sub_word};
 use fuel_asm::PanicReason;
-use fuel_crypto::Hasher;
+use fuel_asm::PanicReason::ArithmeticOverflow;
+use fuel_crypto::{Hasher, Message, PublicKey, Signature};
 use fuel_types::{Bytes32, Bytes64, Word};
 
 impl<S> Interpreter<S> {
     pub(crate) fn ecrecover(&mut self, a: Word, b: Word, c: Word) -> Result<(), RuntimeError> {
-        if a > VM_MAX_RAM - Bytes64::LEN as Word
-            || b > VM_MAX_RAM - Bytes64::LEN as Word
-            || c > VM_MAX_RAM - Bytes32::LEN as Word
+        let bx = checked_add_word(b, Bytes64::LEN as Word)?;
```

Comment:
We have duplicated this logic in several places.

We can add to `Interpreter` two new methods, `read_as` and `write`.

`read_as` will be generic and accept types that implement the `as_ref_unchecked` method(we need to move this method to the trait).

You will be able to write code like:

```rust
let signature: Signature = self.read_as(b)?;
let message: Message = self.read_as(c)?;
```

Reply:
created issue #222